src/js/components/returns/inbound/AddItemsPage.jsx

```diff
@@ -231,22 +210,92 @@ class AddItemsPage extends Component {
     return errors;
   }
 
-  confirmSave(onConfirm) {
-    confirmAlert({
-      title: this.props.translate('react.inboundReturns.message.confirmSave.label', 'Confirm save'),
-      message: this.props.translate(
-        'react.inboundReturns.confirmSave.message',
-        'Are you sure you want to save? There are some lines with empty or zero quantity, those lines will be deleted.',
-      ),
-      buttons: [
-        {
-          label: this.props.translate('react.default.yes.label', 'Yes'),
-          onClick: onConfirm,
-        },
-        {
-          label: this.props.translate('react.default.no.label', 'No'),
-        },
-      ],
+  confirmValidationErrorOnExit() {
```

Comment:
I regret that this component is not a functional one :cry:
When it is functional we will be able to move all of those confirmation alerts to a hook, and make this component a little bit shorter (those alerts have close to 100 lines). Maybe other developers may have any idea how we can make this part of the code a little bit cleaner.

Reply:
> Maybe other developers may have any idea how we can make this part of the code a little bit cleaner.

I can see that the only difference between those are either button labels or title/message. 
We wouldn't achieve much by creating a generic function that would accept the title, message, buttons, because the amount of the props we'd have to provide would probably be the same, but I would have to see it after the refactor.

A second approach might be to have one alert method, and we could potentially have a map that would store the labels and we could access it depending on an alert key, e.g.:
```js
const Alerts = {
  VALIDATION_ON_EXIT: 'VALIDATION_ON_EXIT',
  ...
};

const alertsMap = {
  [Alerts.VALIDATION_ON_EXIT]: {
    title: '...'
    message: '...',
    resolveButtonLabel: '...',
    rejectButtonLabel: '...',
  }
}

runConfirmAlert(key) {
  const labels = alertsMap[key];
  return new Promise((resolve) => {
     confirmAlert({
        title: labels?.title,
        message: labels?.message,
        willUnmount: () => resolve(false),
        buttons: [
              {
                label: labels?.resolveButtonLabel
                onClick: () => resolve(true),
              },
              {
                label: labels?.rejectButtonLabel
                onClick: () => resolve(false),
              },
        ],
    });
  });
}
```

Reply:
I decided not to proceed with it, I feel like generalizing it will make the code a bit less maintainable.
I know these methods are very similar but I think we will benefit more if we leave it separated as we have it now.

Reply:
@drodzewicz OK. I've just wanted to show @alannadolny that there is a possibility to make it generic if we would want to. I'm fine with leaving it, as this component anyway needs to be rewritten anytime soon hopefully, but on the other hand I would not agree that my version makes it difficult to maintain.